### PR TITLE
GMOS pause observation

### PR DESCRIPTION
**Pull request by jluhrs**
Implemented the observation pause for GMOS.

### CODE REVIEW

File: modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala [hunk 1]
```diff
@@ -47,9 +47,9 @@ final case class Execution(execution: List[Action]) {
     * If the index doesn't exist, `Current` is returned unmodified.
     */
   def mark(i: Int)(r: Result): Execution =
-    Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = actionStateFromResult(r)), execution).getOrElse(execution))
+    Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = actionStateFromResult(r)(a.state)), execution).getOrElse(execution))
 
-  def start(i: Int): Execution = Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = Action.Started), execution).getOrElse(execution))
+  def start(i: Int): Execution = Execution(PLens.listNthPLens[Action](i).modg(a => a.copy(state = a.state.copy(runState = Action.Started)), execution).getOrElse(execution))
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala]: Is this a `Monocle` lens?

File: modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala [hunk 2]
```diff
@@ -30,10 +30,12 @@ package engine {
   final case class Action(
     kind: ActionType,
     gen: ActionGen,
-    state: Action.ActionState
+    state: Action.State
   )
   object Action {
 
+    final case class State(runState: ActionState, partials: List[PartialVal])
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/package.scala]: Could you add some comments on what the fields are? I have some idea but it's not fully clear

File: modules/edu.gemini.seqexec.engine/src/test/scala/edu/gemini/seqexec/engine/StepSpec.scala [hunk 3]
```diff
@@ -451,17 +451,20 @@ class StepSpec extends FlatSpec {
   private val observeResult = Result.Observed("dummyId")
   private val result = Result.OK(observeResult)
   private val failure = Result.Error("Dummy error")
-  private val actionFailed =  fromTask(ActionType.Undefined, Task(failure)).copy(state = Action.Failed(failure))
+  private val actionFailed =  fromTask(ActionType.Undefined, Task(failure)).copy(state = Action.State(Action.Failed(failure), List()))
   private val action: Action = fromTask(ActionType.Undefined, Task(result))
-  private val actionCompleted: Action = action.copy(state = Action.Completed(observeResult))
+  private val actionCompleted: Action = action.copy(state = Action.State(Action.Completed(observeResult), List()))
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.engine/src/test/scala/edu/gemini/seqexec/engine/StepSpec.scala]: you can use `Nil` instead of `List()`

**Comment by jluhrs** [comment]
The last commit expands the Action state to keep track of partial results.

File: modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/sequence/steps/StepsControlButtons.scala [hunk 4]
```diff
@@ -68,20 +68,26 @@ object StepsControlButtons {
...
+    Callback(SeqexecCircuit.dispatch(RequestObsPause(id, stepId)))
+
   def handleStop(id: SequenceId, stepId: Int): ScalazReact.ReactST[CallbackTo, State, Unit] =
     ST.retM(requestStop(id, stepId)) >> ST.mod(_.copy(stopRequested = true, abortRequested = true)).liftCB
 
   def handleAbort(id: SequenceId, stepId: Int): ScalazReact.ReactST[CallbackTo, State, Unit] =
     ST.retM(requestAbort(id, stepId)) >> ST.mod(_.copy(abortRequested = true, stopRequested = true)).liftCB
 
+  def handleObsPause(id: SequenceId, stepId: Int): ScalazReact.ReactST[CallbackTo, State, Unit] =
+    ST.retM(requestObsPause(id, stepId)) >> ST.mod(_.copy(stopRequested = true, abortRequested = true)).liftCB
```
Comment (cquiroz) [inline modules/edu.gemini.seqexec.web/edu.gemini.seqexec.web.client/src/main/scala/edu/gemini/seqexec/web/client/components/sequence/steps/StepsControlButtons.scala]: neat

File: modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala (hunk 1 above)
Comment (jluhrs) [reply modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/Execution.scala]: I don't know. I just copied the code from `mark`.

**Pull request merged**